src/hooks/useSyncChainQuery.ts

```diff
@@ -31,8 +41,23 @@ export default function useSyncChainQuery() {
   const [searchParams, setSearchParams] = useSearchParams()
 
   useEffect(() => {
-    if (isActive && urlChainId && chainId !== urlChainId) {
+    // Change a user's chain on pageload if the connected chainId does not match the query param chain
+    if (isActive && urlChainId && chainIdRef.current === chainId && chainId !== urlChainId) {
       selectChain(urlChainId)
     }
-  }, [urlChainId, selectChain, searchParams, setSearchParams, isActive, chainId])
+    // If a user has a connected wallet and has manually changed their chain, delete the old chain query param
+    else if (account && chainIdRef.current !== chainId && chainId !== urlChainId) {
+      if (isSupportedChain(chainId)) {
+        searchParams.set('chain', CHAIN_IDS_TO_NAMES[chainId])
+      } else {
+        searchParams.delete('chain')
+      }
+      setSearchParams(searchParams)
+    }
+
+    // If a user has a connected wallet and the chainId matches the query param chain, update the chainIdRef
+    if (isActive && chainId === urlChainId) {
+      chainIdRef.current = urlChainId
+    }
```

Comment:
Does this _need_ to be in a `useEffect`?

Reply:
Yes, we don't want this triggering outside of one of the dependencies changing, which can cascade to a user's chain being accidentally changed